tests/infra/checker.py

```diff
@@ -40,35 +40,41 @@ def __call__(self, rpc_result, result=None, error=None):
 
 
 def check_can_progress(node, timeout=3):
+    # Check that a write transaction issued on one node is eventually
+    # committed by the service
+    with node.client("user0") as uc:
+        submitted_seqno = uc.post(
+            "/app/log/private", {"id": 42, "msg": "Hello world"}
+        ).seqno
```

Comment:
I don't think we should be checking only `seqno` here. While its correct as long as nothing else is talking to the service, it feels like we're leaving a bug to bite us in future (when _this_ thing never happened but something _else_ grabbed the same seqno in a new term, did we really make progress or just get a signature from a new primary? I guess if its committed that's _some_ progress, but still misleading). If we want to match what the comment above says, I think this should be using `/node/tx` rather than `/node/commit`.

Reply:
I agree, and the more I think about it, the more I think we should think of our usage of `/commit` critically. It's easy to misuse.

Reply:
Good point. We already have the machinery to wait for a commit for a specific `TxID` so I'll update the code to use this instead.